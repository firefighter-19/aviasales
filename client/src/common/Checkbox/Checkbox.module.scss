.checkbox__label {
    display: flex;
    align-items: center;
    margin: 5px 0;
    cursor: pointer;
}

.customCheckbox {
    margin-right: 4px;
    cursor: pointer;
}