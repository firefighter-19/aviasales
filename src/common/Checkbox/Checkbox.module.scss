.checkbox__label {
    display: flex;
    align-items: center;
    margin: 5px 0;
}

.customCheckbox {
    margin-right: 4px;
}