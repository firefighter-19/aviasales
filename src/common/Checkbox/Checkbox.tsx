import React from 'react';

import styles from './CheckBox.module.scss';

interface CheckBoxProps {
    id: string;
    checked: boolean;
    handleChangeCheckBox?(): void;
}

export function CheckBox({ id, checked, handleChangeCheckBox }: CheckBoxProps): JSX.Element {
    return (
        <label htmlFor={id}>
            <input
                type="checkbox"
                className={styles.customCheckbox}
                id={id}
                name="checkbox"
                checked={checked}
                onChange={handleChangeCheckBox}
            />
        </label>
    );
}
