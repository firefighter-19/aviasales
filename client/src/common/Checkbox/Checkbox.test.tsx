import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Checkbox } from './Checkbox';

Enzyme.configure({ adapter: new Adapter() });

describe('Checkbox component testing', () => {
    it("It should return 'Checkbox' component", () => {
        const props = { id: '1', text: 'Без пересадок', checked: false };
        const handleChangeCheckBox = jest.fn();
        const checkbox = shallow(
            <Checkbox
                id={props.id}
                text={props.text}
                checked={props.checked}
                handleChangeCheckBox={handleChangeCheckBox}
            />,
        );
        const checkboxContainer = checkbox.find('.checkbox__container');
        expect(checkboxContainer.length).toBe(1);
    });
    it('It should return Checkboxes components due to checkbox array length', () => {
        const checkboxes = [
            { id: '1', text: 'Без пересадок', checked: false, stops: 0 },
            { id: '3', text: '1 пересадки', checked: false, stops: 1 },
            { id: '2', text: '2 пересадка', checked: false, stops: 2 },
            { id: '4', text: '3 пересадки', checked: false, stops: 3 },
        ];
        const handleChangeCheckBox = jest.fn();
        const checkboxesArray = checkboxes.map((checkbox) => {
            return shallow(
                <Checkbox
                    id={checkbox.id}
                    text={checkbox.text}
                    checked={checkbox.checked}
                    handleChangeCheckBox={handleChangeCheckBox}
                />,
            );
        });
        expect(checkboxesArray).toMatchSnapshot();
    });
    it("It should change 'checked' in checkbox", () => {
        const props = { id: '4', text: '3 пересадки', checked: true, stops: 3 };
        const handleChangeCheckBox = jest.fn();
        const checkbox = shallow(
            <Checkbox
                id={props.id}
                text={props.text}
                checked={props.checked}
                handleChangeCheckBox={handleChangeCheckBox}
            />,
        );
        checkbox.find('input').forEach((node) => {
            expect(node.props().checked).toEqual(true);
        });
    });
});
