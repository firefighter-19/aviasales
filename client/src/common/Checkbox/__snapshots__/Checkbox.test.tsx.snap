// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component testing It should return Checkboxes components due to checkbox array length 1`] = `
Array [
  ShallowWrapper {},
  ShallowWrapper {},
  ShallowWrapper {},
  ShallowWrapper {},
]
`;
